packages/generators/generate-plop/plopfile.js

```diff
@@ -83,9 +87,60 @@ module.exports = function(plop) {
     actions: [
       {
         type: 'add',
-        path: join(process.cwd(), 'config/policies/{{id}}.js'),
+        path: join(rootDir, 'config/policies/{{id}}.js'),
         templateFile: 'templates/policy.js.hbs',
       },
     ],
   });
+
+  // Plugin generator
+  plop.setGenerator('plugin', {
+    description: 'Generate a basic plugin',
+    prompts: [
+      {
+        type: 'input',
+        name: 'id',
+        message: 'Plugin name',
+      },
+    ],
+    actions: data => {
+      fs.copySync(join(__dirname, 'files', 'admin'), join(rootDir, 'plugins', data.id, 'admin'));
+
+      return [
+        {
+          type: 'add',
+          path: join(rootDir, 'plugins/{{id}}/services/{{id}}.js'),
+          templateFile: 'templates/service.js.hbs',
+        },
+        {
+          type: 'add',
+          path: join(rootDir, 'plugins/{{id}}/controllers/{{id}}.js'),
+          templateFile: 'templates/controller.js.hbs',
+        },
+        {
+          type: 'add',
+          path: join(rootDir, 'plugins/{{id}}/config/routes.json'),
+          templateFile: 'templates/routes.json.hbs',
+        },
+        {
+          type: 'add',
+          path: join(rootDir, 'plugins/{{id}}/README.md'),
+          templateFile: 'templates/README.md.hbs',
+        },
+        {
+          type: 'add',
+          path: join(rootDir, 'plugins/{{id}}/package.json'),
+          templateFile: 'templates/plugin-package.json.hbs',
+        },
+        {
+          destination: join(rootDir, 'plugins/{{id}}'),
+          type: 'addMany',
+          templateFiles: 'templates/.*',
+          globOptions: {
+            dot: true,
+          },
+        },
```

Comment:
I find it odd that these are in the templates folder even though they're not handlebars files and they're not dynamic.

Why not putting them with the rest in the files directory? I could imagine a structure like:

- files
    - plugin
        - .editorconfig
        - ...all plugin dot files here
        - admin
            - index.js
            - ...all the plugin admin react files here

This way we could copy the entire `files/plugin` folder. What do you think?

Reply:
sounds good to me! 👍